src/jobserver-win32.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <windows.h>
+#include <cassert>
+
+#include "jobserver.h"
+#include "util.h"
+
+Jobserver::Jobserver() : token_count_(0), sem_(NULL)
+{}
+
+void Jobserver::Init() {
+  assert(sem_ == NULL);
+
+  if (!ParseJobserverAuth("sem")) {
+    return;
+  }
+
+  const char *name = jobserver_name_.c_str();
+
+  sem_ = OpenSemaphore(SEMAPHORE_ALL_ACCESS, false, name);
+  if (sem_ == NULL) {
```

Comment:
nit: use `sem == INVALID_HANDLE_VALUE` here and below.

Reply:
(just to clarify: INVALID_HANDLE_VALUE is really ~0, not 0, so this will not detect invalid semaphore names correctly :-/)

Reply:
Done

Reply:
Documentation stated that `OpenSemaphore()` (and `CreateSemaphore`) return `NULL` on failures. And my recently added tests confirm that.

Reply:
Oh, my bad, sorry, this is kind of unexpected from the Win32 API. Thanks for correcting me on this.